app/controllers/api/subcollections/vms.rb

```diff
@@ -27,6 +20,22 @@ def vms_query_resource(object)
           vm.as_json.merge(conflictless_hash)
         end
       end
+
+      private
+
+      def create_vm_attributes_hash(vm_attrs, vm)
+        vm_attrs.each_with_object({}) do |attr, hash|
+          hash[attr] = vm.public_send(attr.to_sym) if vm.respond_to?(attr.to_sym)
+        end.compact
+      end
+
+      def create_vm_decorators_hash(vm_decorators, vm)
+        decorated = vm.decorator_class? ? vm.decorate : nil
+        return if decorated.nil?
```

Comment:
Just wondering now if `vm.decorator_class?` will ever return `false`? I don't think it will, but if it can I don't believe we have test coverage. So it might be good to either remove the unnecessary code, or prove that it's needed by providing an example. Thanks!